lib/asyn-thread.c

```diff
@@ -403,6 +407,15 @@ static void destroy_async_data(struct Curl_async *async)
 
       free(async->os_specific);
     }
+#ifdef HAVE_SOCKETPAIR
+    /*
+     * ensure CURLMOPT_SOCKETFUNCTION fires CURL_POLL_REMOVE
+     * before the FD is invalidated to avoid EBADF on EPOLL_CTL_DEL
+     */
+    if(conn)
+      Curl_multi_closed(conn->data, sock_rd);
+    sclose(sock_rd);
```

Comment:
small observation, should we invalidate tsd->sock_pair[0] after closing the socket so that it is not accessed mistakenly and result into issues ?

Reply:
I don't think that a big risk though, do you? This the destroy function and the init function sets them up...

Reply:
yeah, my bad...I thought we moved the read socket close to resolver thread.
Sorry for the confusion. !!